src/components/containers/DraggablePanel/DraggablePanel.test.tsx

```diff
@@ -0,0 +1,213 @@
+    expect(panelFromDataTestId.style.transform).toEqual(
+      `translate(${destinationCoordinates.clientX}px,${destinationCoordinates.clientY}px)`
+    );
+
+    expect(panelFromDataTestId).toBeTruthy();
+    expect(handleOnDragComplete).toHaveBeenCalled();
+  });
+
+  test("you can open and close panels", async () => {
+    const defaultPosition = { x: 50, y: 50 };
+
+    function ToggleButtonAndDraggablePanel() {
+      const [panelIsOpen, setPanelIsOpen] = useState(true);
+      return (
+        <>
+          <button onClick={() => setPanelIsOpen((isOpen) => !isOpen)}>
+            Toggle Filters Panel
+          </button>
+          <DraggablePanel
+            defaultPosition={defaultPosition}
+            isOpen={panelIsOpen}
+            panelTitle="My Filters"
+            onDragComplete={() => {}}
+            onPanelDismiss={() => setPanelIsOpen(false)}
+            showPanelTitle
+          >
+            <p>I might be here or I might be gone</p>
+          </DraggablePanel>
+        </>
+      );
+    }
+
+    render(
+      <>
+        <ToggleButtonAndDraggablePanel />
+        <DraggablePanel
+          defaultPosition={defaultPosition}
+          isOpen
+          panelTitle="My Extra Ordinary Data"
+          onDragComplete={() => {}}
+          onPanelDismiss={() => {}}
+          showPanelTitle
+        >
+          <p>I will be with you forever.</p>
+        </DraggablePanel>
+      </>
+    );
+
+    expect(
+      screen.getByText("I might be here or I might be gone")
+    ).toBeVisible();
+
+    const closePanel = screen.getByText("Close My Filters");
+    fireEvent.click(closePanel);
+
+    expect(
+      screen.queryByText("I might be here or I might be gone")
+    ).not.toBeVisible();
+    expect(screen.queryByText("I will be with you forever.")).toBeVisible();
+
+    fireEvent.click(screen.getByText("Toggle Filters Panel"));
+    expect(
+      screen.getByText("I might be here or I might be gone")
+    ).toBeVisible();
+  });
+  test("panels are layered from most-to-least recently dragged", () => {
+    const panelDefinitionOjects: DraggablePanelProps[] = [
+      "Panel 1",
+      "Panel 2",
+      "Panel 3",
+    ].map((panelTitle) => {
+      return {
+        children: () => <p>Panel Contents</p>,
+        panelTitle,
+        showPanelTitle: true,
+        isOpen: true,
+      };
+    });
+
+    render(<StackOrderingKeeper draggablePanelProps={panelDefinitionOjects} />);
+
+    const dragMeFirst = screen.getByText("Panel 3");
+    const dragMeMiddle = screen.getByText("Panel 2");
+    const dragMeLast = screen.getByText("Panel 1");
+
+    drag(dragMeFirst, { clientX: 50, clientY: 50 });
+    drag(dragMeMiddle, { clientX: 60, clientY: 60 });
+    drag(dragMeLast, { clientX: 70, clientY: 70 });
+
+    /**
+     * Asserting on z-index values makes this test brittle to refactoring. Is there
+     * another way to programmatically determine stacking order?
+     */
```

Comment:
If we stacked two panels exactly atop each other, would something like expecting the top one to be visible and the bottom one not to be visible be viable? Not sure if being "visible" means i can actually read it or it's just on the screen somewhere

Reply:
That'd be dope! 

I think the problem is that`toBeVisible()` is a lot less magical. It comes from `jest-dom` and we're referring to visibility in terms of the [DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model). While I suppose you could get the `x` and `y` coordinates of two elements and see if one is obscuring the other, this "matcher" is much less imaginative:

> `toBeVisible()` allows you to check if an element is currently visible to the user.
> 
> An element is visible if all the following conditions are met:
> 
> it is present in the document
> it does not have its css property display set to none
> it does not have its css property visibility set to either hidden or collapse
> it does not have its css property opacity set to 0
> its parent element is also visible (and so on up to the top of the DOM tree)
> it does not have the [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute
> if `<details />` has the open attribute

```html
<div data-testid="zero-opacity" style="opacity: 0">Zero Opacity Example</div>
<div data-testid="visibility-hidden" style="visibility: hidden">
  Visibility Hidden Example
</div>
<div data-testid="display-none" style="display: none">Display None Example</div>
<div style="opacity: 0">
  <span data-testid="hidden-parent">Hidden Parent Example</span>
</div>
<div data-testid="visible">Visible Example</div>
<div data-testid="hidden-attribute" hidden>Hidden Attribute Example</div>
<details>
  <summary>Title of hidden text</summary>
  Hidden Details Example
</details>
<details open>
  <summary>Title of visible text</summary>
  <div>Visible Details Example</div>
</details>
```

https://github.com/testing-library/jest-dom#tobevisible

I guess what I meant in my comment is that [it's possible to have stacking situations that don't involve `z-index`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/Stacking_without_z-index).